cunumeric/eager.py

```diff
@@ -298,7 +300,8 @@ def to_deferred_array(self) -> DeferredArray:
                         shape=self.shape,
                     )
                 else:
-                    self.deferred = self.runtime.find_or_create_array_thunk(  # type: ignore # noqa E501
+                    assert self.runtime is not None
```

Comment:
Is mypy unable to deduce this? The `runtime` attribute is set on the superclass's constructor `NumPyThunk.__init__`, which takes a non-`None` reference and sets it unconditionally to `self.runtime`.

Reply:
It does https://github.com/nv-legate/cunumeric/pull/716/commits/98bde2c32e223aa8c415d731e07156fde852780d